### ISSUE TITLE
Can you please port this fix in zip extraction?

### ISSUE THREAD

**Issue by ale5000-git** [rmyorston/busybox-w32#67]
Can you please port this [fix](https://git.busybox.net/busybox/commit/?id=e3c4db8b396e955ca01648ba51c0fd093f3cab56)?

Ref: https://bugs.busybox.net/show_bug.cgi?id=9536

**Comment by rmyorston** [comment]
The upstream fix doesn't work.  I've reopened the bug but it hasn't yet been corrected.

In the meantime I've applied a local fix that seems to handle all the zip files I've thrown at it.

**Comment by ale5000-git** [comment]
Thanks :)

I have noticed that all apk files (they are zip files) created by the latest Android SDK have this inconsistency in extra length.
I want to ask you one thing.
I know it isn't really related to you but I would like to know if it is done on purpose to hide info (like signature or other things) or it is a bug in the Android SDK.
If it is the latter I want to report it where it belong to.

**Comment by rmyorston** [comment]
I'm no expert on the zip file format but it seems that the extra fields in the central directory and the local header can contain different data and therefore have different lengths.  I found at least two other zip files on my system where this was the case.  `zipinfo -v` handled them correctly and reported that different extra data was present in the central directory and local header.  I think the bug was always in BusyBox, not the Android SDK.

I see that the upstream BusyBox bug has been closed again.  Everything should be OK in future.  I won't make a new build of busybox-w32 yet.
